### PR TITLE
initialize scanner method

### PR DESCRIPTION
**Pull request by brianpmaher**
This PR sets up the basics for beginning scanning. We would need to look over the StreamReader class to decide if that's really a good method to use or if we should try storing it in a string buffer or something (the file that is).
- Scanner.initializeScanner() method: takes in a filename string as a parameter.
- Throws exception if file does not end with .mp format
- Fixed error with enum format.. my bad on merging that one. I thought I tested it; but this does fix that.

### CODE REVIEW

File: constants.cs [hunk 1]
```diff
@@ -1,75 +1,79 @@
 using System;
 
-// Contains all constants for our program. Put any large and/or repeated strings here.
-public static class Constants {
-	public const string USAGE_HELP = "Usage: mono YµP.exe <µPascal filename> <output filename>";
+// All token IDs (add more as we come across them)
+public enum TOKENS {
```
Comment (brianpmaher) [inline constants.cs:4]: All changes relating to TOKENS and Constants class were just to move the TOKENS enum to the correct location (above the class and not inside of it; which was throwing some errors)

File: constants.cs [hunk 2]
```diff
@@ -1,75 +1,79 @@
...
+	public const string USAGE_HELP = "Usage: mono YµP.exe <µPascal filename> <output filename>";
 
-		// Special Tokens
-		EOF,			// end-of-file character
-		RUN_COMMENT,	// run-on comment error
-		RUN_STRING,		// run-on string error
-		ERROR,			// token for other scan errors
-	};
+	// Error messages
+	public const string ERROR_FILE_FORMAT = "ERROR: File format does not conform to format: <filename>.mp; ";
```
Comment (brianpmaher) [inline constants.cs]: Added fileFormat exception message.

File: scanner.cs [hunk 3]
```diff
@@ -1,8 +1,20 @@
 using System;
+using System.IO;
 
 public class Scanner {
-	public void initializeScanner(string file_name) {
-
+	private StreamReader __reader;
```
Comment (JesseStephenBrown) [inline scanner.cs]: We may want to try to read byte by byte, that way we can just count bytes read as our fp (easy to reset the fp to where we need to be, also). StreamReader is....meh. Unless you can figure out a way to reset the pointer.

**Comment by JesseStephenBrown** [comment]
+1

**Comment by brianpmaher** [comment]
It's a good idea to make sure all of our text editors/IDEs are setup to use 4 spaces instead of tabs.

**Comment by JesseStephenBrown** [comment]
+1 - stop committing :exclamation:

**Comment by seanrogers2657** [comment]
+1 looks good!

**Pull request merged**